src/OSPSuite.Core/Import/DataFormatParameter.cs

```diff
@@ -0,0 +1,93 @@
+﻿namespace OSPSuite.Core.Import
+{
+   public class ParameterConfiguration
+   {
+      public ParameterConfiguration(object data = null)
+      {
+         Data = data;
+      }
+
+      public object Data { get; private set; }
+
+      public override string ToString()
+      {
+         return Data?.ToString() ?? string.Empty;
+      }
+   }
+
+   public abstract class DataFormatParameter
+   {
+      public string ColumnName { get; set; }
+
+      protected DataFormatParameter(string columnName)
+      {
+         ColumnName = columnName;
+      }
+
+      protected DataFormatParameter() { }
+
+      public virtual bool EquivalentTo(DataFormatParameter other)
+      {
+         return other.GetType() == GetType();
+      }
+   }
+   
+   public class AddGroupByFormatParameter : DataFormatParameter
+   {
+      public AddGroupByFormatParameter() { }
+
+      public AddGroupByFormatParameter(string columnName) : base(columnName)
+      {
+      }
+   }
+
+   public class MetaDataFormatParameter : DataFormatParameter
+   {
+      public MetaDataFormatParameter() { }
+
+      public MetaDataFormatParameter(string columnName, string metaDataId, bool isColumn = true) : base(columnName)
+      {
+         MetaDataId = metaDataId;
+         IsColumn = isColumn;
+      }
+
+      public string MetaDataId { get; set; }
+
+      public bool IsColumn { get; set; }
```

Comment:
When the metadata information (ColumnName) changes it could allow for values coming from the project and not from the excel column. It means that if you modify an existing parameter to point the metadata to a value it also needs to change the IsColumn value. I think it would be better to have as a setter function that would do both together so you cannot break it from the outside, meaning when you set the value you specify if it comes from a column or a value and the setter sets both. The problem is that ColumnName is comming from the base class and do not have a setter method that will take two parameters so we could think this deeply or leave it like it is for now and later iterate on it.

Reply:
Check on AbstractColumnsDataFormat.Paramters